common/src/test/java/com/microsoft/identity/common/BrokerOAuth2TokenCacheTest.java

```diff
@@ -1241,4 +1248,120 @@ public void testClearAll() throws ClientException {
+        final String storeName2 = "test_store_name_2";
+        final IPlatformComponents components = mPlatformComponents;
+
+        // Call getCacheToBeUsed with different storeNames
+        final IAccountCredentialCache cache1 = BrokerOAuth2TokenCache.getCacheToBeUsed(components, storeName1);
+        final IAccountCredentialCache cache2 = BrokerOAuth2TokenCache.getCacheToBeUsed(components, storeName2);
+
+        // Verify both are valid but different instances
+        assertNotNull(cache1);
+        assertNotNull(cache2);
+        assertNotSame("Expected different cache instances for different storeNames", cache1, cache2);
+        assertTrue("Cache should be of type SharedPreferencesAccountCredentialCacheWithMemoryCache",
+                cache1 instanceof SharedPreferencesAccountCredentialCacheWithMemoryCache);
+        assertTrue("Cache should be of type SharedPreferencesAccountCredentialCacheWithMemoryCache",
+                cache2 instanceof SharedPreferencesAccountCredentialCacheWithMemoryCache);
+    }
+
+    @Test
+    public void testCacheInstanceReusedAcrossMultipleBrokerTokenCaches_FlightEnabled() throws Exception {
+        // Enable the flight
+        setFlight();
+
+        final String storeName = getBrokerUidSequesteredFilename(TEST_APP_UID);
+
+        // Create multiple BrokerOAuth2TokenCache instances
+        final BrokerOAuth2TokenCache tokenCache1 = new BrokerOAuth2TokenCache
+                (mPlatformComponents,
+                        TEST_APP_UID,
+                        new NameValueStorageBrokerApplicationMetadataCache(mPlatformComponents));
+        final BrokerOAuth2TokenCache tokenCache2 = new BrokerOAuth2TokenCache(mPlatformComponents, TEST_APP_UID,
+                new NameValueStorageBrokerApplicationMetadataCache(mPlatformComponents));
+
+        // Get the underlying account credential caches
+        final IAccountCredentialCache cache1 = tokenCache1.getCacheToBeUsed(mPlatformComponents, storeName);
+        final IAccountCredentialCache cache2 = tokenCache2.getCacheToBeUsed(mPlatformComponents, storeName);
+
+        // Verify same instance is reused
+        assertNotNull(cache1);
+        assertNotNull(cache2);
+        assertSame(cache1, cache2);
+    }
+
+    @Test
+    public void testFociCacheInstanceReused_FlightEnabled() {
+        // Enable the flight
+        setFlight();
+
+        final String fociStoreName = BROKER_FOCI_ACCOUNT_CREDENTIAL_SHARED_PREFERENCES;
+
+        // Call getCacheToBeUsed multiple times for FOCI cache
+        final IAccountCredentialCache fociCache1 = BrokerOAuth2TokenCache.getCacheToBeUsed(mPlatformComponents, fociStoreName);
+        final IAccountCredentialCache fociCache2 = BrokerOAuth2TokenCache.getCacheToBeUsed(mPlatformComponents, fociStoreName);
+
+        // Verify same FOCI cache instance is reused
+        assertNotNull(fociCache1);
+        assertNotNull(fociCache2);
+        assertSame(fociCache1, fociCache2);
```

Comment:
reset flights manager after test. The instance lives beyond method life and can impact other tests.